doc/nrf/config_and_build/programming.rst

```diff
@@ -1,133 +1,88 @@
+      #. Power on the development kit.
+      #. Make sure you are familiar with :ref:`programming_hw`.
+      #. Program the application to the kit using the following command:
 
-      .. note::
-         To program the nRF52840 Dongle instead of a development kit, skip the following instructions and follow the programming instructions in :ref:`zephyr:nrf52840dongle_nrf52840`.
+         .. code-block:: console
 
-#.    Power on the development kit.
-#.    Program the application to the kit using the following command:
+           west flash --erase
 
-      .. code-block:: console
+         This command erases the full flash memory before programming, which is the recommended approach.
+         If the application depends on other flash memory areas (for example, if it uses the :ref:`zephyr:settings_api` partition where bonding information is stored), erasing the full kit before programming ensures that these areas are updated with the new content.
+         You can also :ref:`program without erasing <programming_params_no_erase>`.
 
-         west flash --erase
+         The ``west flash`` command automatically resets the kit and starts the application.
 
-      This command erases the full flash memory before programming, which is the recommended approach.
-      If the application depends on other flash memory areas (for example, if it uses the :ref:`zephyr:settings_api` partition where bonding information is stored), erasing the full kit before programming ensures that these areas are updated with the new content.
+      For more information on programming using the command line, see :ref:`Flashing <zephyr:west-flashing>` in the Zephyr documentation.
 
-      As an alternative, you can also clear only those flash memory pages that are to be overwritten with the new application.
-      With such approach, the old data in other areas will be retained.
 
-      To erase only the areas of flash memory that are required for programming the new application, use the following command:
+.. _programming_hw:
 
-      .. code-block:: console
+Hardware-specific programming steps
+===================================
 
-         west flash
+Some hardware has platform-specific requirements that must be met before you program the device.
+For additional information, check the :ref:`user guide for the hardware platform <device_guides>` that you are using.
 
-      The ``west flash`` command automatically resets the kit and starts the application.
+For example, if you are working with an nRF9160 DK, you need to select the correct position of the :ref:`board controller <nrf9160_ug_intro>` switch **SW10** before you program the application to your development kit.
+Programming to Thingy:91 also requires a :ref:`similar step <building_pgming>`, but using a different switch (**SW2**).
 
-For more information on building and programming using the command line, see the Zephyr documentation on :ref:`zephyr:west-build-flash-debug`.
+Programming the nRF52840 Dongle
```

Comment:
Do we have any docs for programming custom boards somewhere? If so, that would be relevant as an additional point to make in this section.

Reply:
No, we need them. I'll handle this update separately.